File: PHP/Guest.php
<?php

/**
 * Implements a data structure for Guests of the type ID,Name,Surname,List of Friend's IDs
 */
class Guest
{

    private $id;
    private $name;
    private $surname;
    public $friends;

    /**
     * Guest constructor.
     * @param $id int Guests ID.
     * @param $name string Guests first name.
     * @param $surname string Guests last name.
     * @param $friends int[] List of friends IDs.
     */
    public function __construct($id, $name, $surname, $friends)
    {
        $this->id = $id;
        $this->name = $name;
        $this->surname = $surname;
        $this->friends = $friends;
    }

    /**
     * Get a string of the guests data.
     * @return string "$ID, $NAME, $SURNAME, [$FRIENDS]"
     */
    public function toString()
    {
        return "$this->id, $this->name, $this->surname, " . implode(",", $this->friends);
    }

    /**
     * Remove a friends ID from the list.
     * @param $id int
     */
    public function removeFriend(int $id)
    {
        if (($key = array_search($id, $this->friends)) !== false) {
            unset($this->friends[$key]);

        }
    }

    public function getId(): int
    {
        return $this->id;
    }

    public function getNumberOfFriends()
    {
        return sizeof($this->friends);
    }

}

File: VisualBasic/PartyProblem/Guest.vb
''' <summary>
'''     Implements a data structure for Guests of the type ID, Name, Surname, List of Friends IDs.
''' </summary>
Public Class Guest
    Private ReadOnly _id As Short
    Private ReadOnly _name As String
    Private ReadOnly _surname As String
    Private ReadOnly _friends As List(Of Short)

    Public ReadOnly Property Id As Short
        Get
            Return _id
        End Get
    End Property

    ''' <summary>
    '''     Constructor.
    ''' </summary>
    ''' <param name="id">Guests ID.</param>
    ''' <param name="name">Guests first name.</param>
    ''' <param name="surname">Guests last name.</param>
    ''' <param name="friends">List of friends IDs.</param>
    Public Sub New(id As Short, name As String, surname As String, friends As List(Of Short))
        Me._id = id
        Me._name = name
        Me._surname = surname
        Me._friends = friends
    End Sub

    ''' <summary>
    '''     Remove a friends ID from the list.
    ''' </summary>
    ''' <param name="id">ID of the friend to remove.</param>
    Public Sub RemoveFriend(id As Short)
        Me._friends.Remove(id)
    End Sub

    ''' <summary>
    '''     Get a string version of the guests data.
    ''' </summary>
    ''' <returns>"$ID, $NAME $SURNAME, [$FRIENDS]"</returns>
    Public Overrides Function ToString() As String
        Return Me._id.ToString() + ", " + Me._name + ", " + Me._surname + ", [" + String.Join(", ", Me._friends) +
               "]"
    End Function

    ''' <summary>
    '''     Get the number of the guests friends.
    ''' </summary>
    ''' <returns>Number of friends.</returns>
    Public Function GetNumberOfFriends() As Short
        Return Me._friends.Count
    End Function
End Class